### ISSUE TITLE
GraphPanel widget: log scales don't work in some cases, and panning/scaling with log scales could be improved

### ISSUE THREAD

**Issue by agarny** [opencor/opencor#1412]
If, using the Noble 1962 model that is shipped with OpenCOR, we plot I_K against time, we get this:

![image](https://user-images.githubusercontent.com/602265/30495750-86f02772-9a4d-11e7-84bb-ca72d7966275.png)

From there, if we ask for the Y axis to use a logarithmic scale, we get this:

![image](https://user-images.githubusercontent.com/602265/30495790-9d7ddd90-9a4d-11e7-95a6-80e9dc9fb33f.png)

which in itself is correct, except that it could actually be improved. The original range is [14; 50] while it is [10; 60] when a logarithmic scale is used for the Y axis. Ideally, it would be the same range (so as to maximise the rendering of the graph), i.e.

![image](https://user-images.githubusercontent.com/602265/30495805-ac651f58-9a4d-11e7-826c-33b33dcb4d81.png)

This aside, things really go bad when using a logarithmic scale on the X axis:

![image](https://user-images.githubusercontent.com/602265/30495821-c0e24230-9a4d-11e7-80a2-37bfc7f53dd6.png)

The X range is [1e-100; 1] while in this particular case, we would probably expect it to be [1; 1000] (and [15; 50] for the Y axis – and not [14; 50] since we can’t plot the data point at t=0), i.e.

![image](https://user-images.githubusercontent.com/602265/30495829-cd8992ae-9a4d-11e7-9381-01fede961413.png)

Anyway, it looks like there are a few things to fix and improve here and there when it comes to the log scales.

(Another useful test is to plot I_K against the membrane potential.)

**Comment by agarny** [comment]
Argh, some to plot V against time in N62 will result in the following output:

![screen shot 2017-10-05 at 17 06 46](https://user-images.githubusercontent.com/602265/31234628-b1b1e0de-a9ef-11e7-9524-f488ff2f6ef3.png)

However, to double click on every single graph panel results in the following output:

![screen shot 2017-10-05 at 17 06 56](https://user-images.githubusercontent.com/602265/31234828-3942c05e-a9f0-11e7-941e-cacd12a95682.png)

In other words, although the 2nd and 3rd graph panels (with an x-log and y-log axis, respectively) look right, their axes' range is actually wrong... So, reopening... :(
